### PR TITLE
Adjusted script command progressbar_npc behavior

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: None

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Allow players to stay attached.
  * Use Work In Progress value to restrict while progress bar is active.

### CODE REVIEW

**Comment by attackjom** [comment]
This has a important in ep17.1 script++

but, I think it will be good if we have optional to detach or attach to character.

**Comment by JohnnyPlayy** [comment]
It is also very important in 17.2

File: src/map/script.cpp [hunk 1]
```diff
@@ -21014,6 +21014,7 @@ BUILDIN_FUNC(progressbar)
  * progressbar_npc "<color>",<seconds>{,<"NPC Name">};
  */
 BUILDIN_FUNC(progressbar_npc){
+	map_session_data *sd;
```
Comment (Atemo) [inline src/map/script.cpp:21017]:
```suggestion
	map_session_data *sd = map_id2sd(st->rid);
```

File: src/map/script.cpp [hunk 2]
```diff
@@ -21042,8 +21043,10 @@ BUILDIN_FUNC(progressbar_npc){
 			return SCRIPT_CMD_FAILURE;
 		}
 
-		// detach the player
-		script_detach_rid(st);
+		if (script_rid2sd(sd)) { // Player attached - keep them from doing other things
```
Comment (Atemo) [inline src/map/script.cpp:21046]:
```suggestion
		if (sd) { // Player attached - keep them from doing other things
```

File: src/map/script.cpp [hunk 3]
```diff
@@ -21055,6 +21058,11 @@ BUILDIN_FUNC(progressbar_npc){
 	// Second call(by timer after sleeping time is over)
 	} else {
 		// Continue the script
+		if (script_rid2sd(sd)) { // Player attached - remove restrictions
```
Comment (Atemo) [inline src/map/script.cpp:21061]:
```suggestion
		if (sd) { // Player attached - remove restrictions
```

**Pull request merged**